schema/table.go

```diff
@@ -520,6 +520,9 @@ func (t *Table) TableNames() []string {
 }
 
 func (t *Table) Copy(parent *Table) *Table {
+	if t == nil {
+		return nil
+	}
```

Comment:
Is this change for testing? Could we move the `if t == nil` logic to the test instead? I'm just wary that adding this here might hide future bugs

Reply:
Definitely, we can move it, but currently it panics if you try to copy a nil table...

Reply:
> currently it panics if you try to copy a nil table...

Sounds reasonable to me, you shouldn't be trying to copy a nil table :)